src/validation/ValidationSchema.js

```diff
@@ -27,27 +27,22 @@ const validationSchema = yup.object({
       })
       /* eslint-disable func-names */
       .test('test', 'Minimum must be less than the maximum', function (value) {
-        const { type, maxAmount } = this.parent;
-        if (
-          type === AmountType.Range &&
-          value > 0 &&
-          maxAmount > 0 &&
-          value >= maxAmount
-        )
+        const { type, max } = this.parent;
+        if (type === AmountType.Range && value > 0 && max > 0 && value >= max)
           return false;
         return true;
       })
       .test(
         'test',
         'Range amount must have a minimum and/or a maximum',
         function (value) {
-          const { type, maxAmount } = this.parent;
-          if (type === AmountType.Range && value === 0 && maxAmount === 0)
+          const { type, max } = this.parent;
+          if (type === AmountType.Range && value === 0 && max === 0)
             return false;
           return true;
         }
       ),
```

Comment:
Does this work?
```jsx
      .when('type', {
        is: AmountType.Range,
        then: yup
          .test('test', 'Minimum must be less than the maximum',
            (min, {parent}) => min === 0 || parent.max > 0 && min < max
          )
          .test(
            'test',
            'Range amount must have a minimum and/or a maximum',
            (min, {parent}) =>  min > 0 || parent.max > 0
          )
      }),
```

I think it's better to leave the tests only for the Range type. Also, I just learned that the function you pass to [test()](https://github.com/jquense/yup#mixedtestname-string-message-string--function-test-function-schema) actually accepts the **this** context as the second parameter so you can write them more succinctly and get rid of the lint error.

Reply:
woah yeah it worked! I just had to add `|| parent.max === 0` for the first test so that It could handle the min being filled while the max stayed empty.

Reply:
Ahh it probably should be `(min, {parent}) min === 0 || parent.max === 0 || min < max` then. Either one of them is 0 or min is less than max.